### ISSUE TITLE
Use /usr/bin/env in Bash shebangs

### ISSUE THREAD

**Issue by nullishamy** [williamboman/mason.nvim#308]
### I've made an initial investigation into the error myself

- [X] Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

When mason writes bash shebangs (https://github.com/williamboman/mason.nvim/blob/main/lua/mason-core/installer/context.lua#L242), it assumes there is a bash install located at `/bin/bash`. Whilst this is true on most systems, there are many systems where this is not true.

Instead of this, mason should use an env shebang, `#!/usr/bin/env bash`, which will query the environment to locate a bash install.

### Neovim version (>= 0.7)

NVIM v0.7.0
Build type: Release
LuaJIT 2.1.0-beta3

### Operating system/version

Linux nixon 5.15.59 #1-NixOS SMP Wed Aug 3 10:03:56 UTC 2022 x86_64 GNU/Linux

### Affected packages

mason

### Actual behavior

Mason writes an invalid bash shebang, causing errors.

### Expected behavior

Mason writes a valid shebang.

### Installation log

```Text
[INFO  Wed 17 Aug 2022 23:31:47 BST] ...acker/start/mason.nvim/lua/mason-core/installer/init.lua:110: Executing installer for Package(name=lua-language-server)
[DEBUG Wed 17 Aug 2022 23:31:47 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:70: fs: mkdirp /home/amy/.local/share/nvim/mason/.packages/lua-language-server
[DEBUG Wed 17 Aug 2022 23:31:47 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="gh", spawn_opts={
  args = { "api", "repos/sumneko/vscode-lua/releases" }
}
[DEBUG Wed 17 Aug 2022 23:31:47 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85474
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="npm", spawn_opts={
  args = { "outdated", "--json", "bash-language-server" },
  cwd = "/home/amy/.local/share/nvim/mason/packages/bash-language-server"
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85482
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="npm", spawn_opts={
  args = { "outdated", "--json", "eslint_d" },
  cwd = "/home/amy/.local/share/nvim/mason/packages/eslint_d"
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85483
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="python", spawn_opts={
  args = { "-m", "pip", "list", "--outdated", "--format=json" },
  cwd = "/home/amy/.local/share/nvim/mason/packages/gitlint",
  env = { "NVIM_LISTEN_ADDRESS=<redacted>", "XCURSOR_PATH=<redacted>", "DBUS_SESSION_BUS_ADDRESS=<redacted>", "NIX_PROFILES=<redacted>", "TERMINAL=<redacted>", "LSCOLORS=<redacted>", "TERM=<redacted>", "ALACRITTY_LOG=<redacted>", "XDG_SESSION_TYPE=<redacted>", "QTWEBKIT_PLUGIN_PATH=<redacted>", "GEM_HOME=/nix/store/8cqdh6am33wn6y43bp2zvl99xqisd78i-neovim-ruby-env/lib/ruby/gems/2.7.0", "LANG=<redacted>", "GDMSESSION=<redacted>", "XAUTHORITY=<redacted>", "NIXPKGS_CONFIG=<redacted>", "VIMRUNTIME=<redacted>", "LUA_CPATH=<redacted>", "XDG_CURRENT_DESKTOP=<redacted>", "NVIM_LOG_FILE=<redacted>", "MOZ_PLUGIN_PATH=<redacted>", "PAGER=<redacted>", "VIM=<redacted>", "SHLVL=<redacted>", "ALACRITTY_SOCKET=<redacted>", "LIBEXEC_PATH=<redacted>", "USER=<redacted>", "TERMINFO_DIRS=<redacted>", "__HM_ZSH_SESS_VARS_SOURCED=<redacted>", "GTK_PATH=<redacted>", "NO_AT_BRIDGE=<redacted>", "NIX_PATH=<redacted>", "XDG_SEAT_PATH=<redacted>", "SHELL=<redacted>", "EDITOR=<redacted>", "COLORTERM=<redacted>", "XDG_VTNR=<redacted>", "WINDOWID=<redacted>", "LESS=<redacted>", "XDG_CONFIG_DIRS=<redacted>", "KDEDIRS=<redacted>", "LOCALE_ARCHIVE=<redacted>", "XDG_SESSION_CLASS=<redacted>", "TZDIR=<redacted>", "XDG_DATA_DIRS=<redacted>", "LESSOPEN=<redacted>", "LD_LIBRARY_PATH=<redacted>", "__ETC_PROFILE_DONE=<redacted>", "XDG_SESSION_DESKTOP=<redacted>", "NIX_USER_PROFILE_DIR=<redacted>", "GTK2_RC_FILES=<redacted>", "NVIM_SYSTEM_RPLUGIN_MANIFEST=<redacted>", "HOME=<redacted>", "SSH_ASKPASS=<redacted>", "INFOPATH=<redacted>", "PWD=<redacted>", "LUA_PATH=<redacted>", "LOGNAME=<redacted>", "LS_COLORS=<redacted>", "XDG_SESSION_ID=<redacted>", "XDG_SESSION_PATH=<redacted>", "DISPLAY=<redacted>", "XDG_SEAT=<redacted>", "__HM_SESS_VARS_SOURCED=<redacted>", "__NIXOS_SET_ENVIRONMENT_DONE=<redacted>", "LESSKEYIN_SYSTEM=<redacted>", "DESKTOP_SESSION=<redacted>", "XDG_GREETER_DATA_DIR=<redacted>", "MYVIMRC=<redacted>", "QT_PLUGIN_PATH=<redacted>", "XDG_RUNTIME_DIR=<redacted>", "OLDPWD=<redacted>", "LOCALE_ARCHIVE_2_27=<redacted>", "I3SOCK=<redacted>", "PATH=/home/amy/.local/share/nvim/mason/packages/gitlint/venv/bin:/home/amy/.local/share/nvim/mason/bin:/run/wrappers/bin:/home/amy/.nix-profile/bin:/etc/profiles/per-user/amy/bin:/nix/var/nix/profiles/default/bin:/run/current-system/sw/bin:/home/amy/.zsh/plugins/zsh-autosuggestions:/home/amy/.zsh/plugins/zsh-syntax-highlighting:/nix/store/8cqdh6am33wn6y43bp2zvl99xqisd78i-neovim-ruby-env/bin" }
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85484
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="go", spawn_opts={
  args = { "list", "-json", "-m", "golang.org/x/tools/gopls@latest" },
  cwd = "/home/amy/.local/share/nvim/mason/packages/gopls"
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85495
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="luarocks", spawn_opts={
  args = { "list", "--outdated", "--tree", "/home/amy/.local/share/nvim/mason/packages/luacheck", "--porcelain" }
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85500
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85495 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="go", spawn_opts={
  args = { "version", "-m", "gopls" },
  cwd = "/home/amy/.local/share/nvim/mason/packages/gopls"
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85517
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85517 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="gh", spawn_opts={
  args = { "api", "repos/OmniSharp/omnisharp-roslyn/releases" }
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85525
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85474 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:48 BST] ...art/mason.nvim/lua/mason-core/managers/github/client.lua:82: Resolved latest version repo="sumneko/vscode-lua", tag_name="v3.5.3"
[DEBUG Wed 17 Aug 2022 23:31:48 BST] ...te/pack/packer/start/mason.nvim/lua/mason-core/fetch.lua:32: Fetching URL "https://github.com/sumneko/vscode-lua/releases/download/v3.5.3/vscode-lua-v3.5.3-linux-x64.vsix"
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="wget", spawn_opts={
  args = { "--header='User-Agent: mason.nvim (+https://github.com/williamboman/mason.nvim)'", "-nv", "-O", "/home/amy/.local/share/nvim/mason/.packages/lua-language-server/archive.zip", "--method=GET", "https://github.com/sumneko/vscode-lua/releases/download/v3.5.3/vscode-lua-v3.5.3-linux-x64.vsix" }
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85575
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85483 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="npm", spawn_opts={
  args = { "outdated", "--json", "@fsouza/prettierd" },
  cwd = "/home/amy/.local/share/nvim/mason/packages/prettierd"
}
[DEBUG Wed 17 Aug 2022 23:31:48 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85576
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85525 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:49 BST] ...art/mason.nvim/lua/mason-core/managers/github/client.lua:82: Resolved latest version repo="OmniSharp/omnisharp-roslyn", tag_name="v1.39.1"
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="gh", spawn_opts={
  args = { "api", "repos/johnnymorganz/stylua/releases" }
}
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85595
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85500 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="npm", spawn_opts={
  args = { "outdated", "--json", "typescript-language-server" },
  cwd = "/home/amy/.local/share/nvim/mason/packages/typescript-language-server"
}
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85602
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85482 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85595 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:49 BST] ...art/mason.nvim/lua/mason-core/managers/github/client.lua:82: Resolved latest version repo="johnnymorganz/stylua", tag_name="v0.14.2"
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85602 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:49 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85484 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85575 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="unzip", spawn_opts={
  args = { "-d", ".", "archive.zip" },
  cwd = "/home/amy/.local/share/nvim/mason/.packages/lua-language-server"
}
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85613
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85613 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:58: fs: unlink /home/amy/.local/share/nvim/mason/.packages/lua-language-server/archive.zip
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:91: fs: write_file /home/amy/.local/share/nvim/mason/.packages/lua-language-server/lua-language-server
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:116: Spawning cmd="chmod", spawn_opts={
  args = { "+x", "lua-language-server" },
  cwd = "/home/amy/.local/share/nvim/mason/.packages/lua-language-server"
}
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:162: Spawned with pid 85614
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85614 exited with exit_code=0, signal=0
[DEBUG Wed 17 Aug 2022 23:31:50 BST] ...er/start/mason.nvim/lua/mason-core/installer/context.lua:174: Promoting cwd "/home/amy/.local/share/nvim/mason/.packages/lua-language-server" to "/home/amy/.local/share/nvim/mason/packages/lua-language-server"
[DEBUG Wed 17 Aug 2022 23:31:50 BST] ...ker/start/mason.nvim/lua/mason-core/installer/linker.lua:21: Unlinking Package(name=lua-language-server)
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:58: fs: unlink /home/amy/.local/share/nvim/mason/bin/lua-language-server
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:46: fs: rmrf /home/amy/.local/share/nvim/mason/packages/lua-language-server
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:64: fs: mkdir /home/amy/.local/share/nvim/mason/packages/lua-language-server
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:83: fs: rename /home/amy/.local/share/nvim/mason/.packages/lua-language-server /home/amy/.local/share/nvim/mason/packages/lua-language-server
[DEBUG Wed 17 Aug 2022 23:31:50 BST] ...ker/start/mason.nvim/lua/mason-core/installer/linker.lua:80: Linking Package(name=lua-language-server)
[DEBUG Wed 17 Aug 2022 23:31:50 BST] ...ker/start/mason.nvim/lua/mason-core/installer/linker.lua:46: Linking bin "lua-language-server" to "../packages/lua-language-server/lua-language-server"
[DEBUG Wed 17 Aug 2022 23:31:50 BST] ...acker/start/mason.nvim/lua/mason-core/installer/init.lua:30: Writing receipt for Package(name=lua-language-server)
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../site/pack/packer/start/mason.nvim/lua/mason-core/fs.lua:91: fs: write_file /home/amy/.local/share/nvim/mason/packages/lua-language-server/mason-receipt.json
[INFO  Wed 17 Aug 2022 23:31:50 BST] ...acker/start/mason.nvim/lua/mason-core/installer/init.lua:142: Installation succeeded for Package(name=lua-language-server)
[DEBUG Wed 17 Aug 2022 23:31:50 BST] .../pack/packer/start/mason.nvim/lua/mason-core/process.lua:148: Job pid=85576 exited with exit_code=0, signal=0
```

### Healthcheck

```Text
mason: require("mason.health").check()
========================================================================
## mason.nvim report
  - OK: neovim version >= 0.7.0
  - OK: **Go**: `go version go1.17.10 linux/amd64`
  - WARNING: **cargo**: not available
  - OK: **luarocks**: `/nix/store/x22zvmanfdrpifibbp4mnji874rvpiz2-luarocks-3.8.0/bin/.luarocks-wrapped 3.8.0`
  - OK: **Ruby**: `ruby 2.7.6p219 (2022-04-12 revision c9c2245c0a) [x86_64-linux]`
  - OK: **RubyGem**: `3.2.26`
  - WARNING: **Composer**: not available
  - WARNING: **PHP**: not available
  - OK: **npm**: `8.11.0`
  - OK: **node**: `v16.16.0`
  - WARNING: **python3**: not available
  - WARNING: **pip3**: not available
  - WARNING: **javac**: not available
  - WARNING: **java**: not available
  - WARNING: **julia**: not available
  - OK: **wget**: `GNU Wget 1.21.3 built on linux-gnu.`
  - OK: **curl**: `curl 7.83.1 (x86_64-pc-linux-gnu) libcurl/7.83.1 OpenSSL/1.1.1q zlib/1.2.12 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libssh2/1.10.0 nghttp2/1.47.0`
  - OK: **gzip**: `gzip 1.12`
  - OK: **tar**: `tar (GNU tar) 1.34`
  - OK: **bash**: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
  - OK: **sh**: `Ok`
  - OK: **python3_host_prog**: `Python 3.9.13`
  - OK: GitHub API rate limit. Used: 9. Remaining: 4991. Limit: 5000. Reset: Thu 18 Aug 2022 00:16:46 BST.
```

**Comment by williamboman** [comment]
Hello! Ah yes of course, will revise

**Comment by williamboman** [comment]
Fixed in #315!
